src/test/java/com/vaadin/flow/ai/formfiller/FormFillerTest.java

```diff
@@ -0,0 +1,41 @@
+package com.vaadin.flow.ai.formfiller;
+
+import com.vaadin.flow.ai.formfiller.services.FormFillerStats;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.internal.UsageStatistics;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.doNothing;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+
+//TODO: Add more tests for FormFiller
+public class FormFillerTest {
```

Comment:
For the FormFillerTest I'd propose do the following:
1. Create a mock of `ChatGPTService/ChatGPTChatCompletionService` with stubbed `createCompletion` and `createChatCompletion` methods.
2. Set up a layout with fields and pass it to Form Filler along with the mock from 1.
3. Capture inputs of `createCompletion` and verify that they are the expected ones, i.e. `ChatGPTService/ChatGPTChatCompletionService` prepare a proper input for AI third-party service.
4. Verify that the Form Filler returns a proper `FormFillerResult` for a stub returned by `createCompletion` and `createChatCompletion` .

Reply:
This is a good idea, thank you. 

I think we can mock those methods, in a similar way as we did in the ChatGPTChatCompletionService Test and ChatGPTServiceTest!

Also probably I can mock in the UI tests as well, as we have some end to end integration test in the demo, maybe we do not need to have the same here.